### PR TITLE
ARROW-9416: [Go] Add testcases for some datatypes

### PR DESCRIPTION
**Pull request by dota17**

### CODE REVIEW

File: go/arrow/compare_test.go [hunk 1]
```diff
@@ -147,6 +150,37 @@ func TestTypeEqual(t *testing.T) {
...
+					Field{Name: "f1", Type: PrimitiveTypes.Uint32, Nullable: true},
+				},
+				index: map[string]int{"f1": 0},
+			},
+			&StructType{
+				fields: []Field{
+					Field{Name: "f1", Type: PrimitiveTypes.Uint32, Nullable: true},
+					Field{Name: "f2", Type: PrimitiveTypes.Uint32, Nullable: true},
+				},
+				index: map[string]int{"f1": 0, "f2": 0},
```
Comment (sbinet) [inline go/arrow/compare_test.go:165]:
```suggestion
				index: map[string]int{"f1": 0, "f2": 1},
```
(I think. this should probably be fixed above as well.)

**Review by sbinet — COMMENTED** [review]
LGTM, modulo the small nitpick.

**Comment by fredgan** [comment]
@sbinet Sorry my mistake. I copied from the above case, so I modify it, too.

Anyway, it's okay now.

**Comment by sbinet** [comment]
probably more mine (mistake) than yours :)

**Review by sbinet — APPROVED** [review]

**Pull request closed without merging**